mp/src/game/server/neo/neo_player.cpp

```diff
@@ -2331,14 +2333,44 @@ bool CNEO_Player::ProcessTeamSwitchRequest(int iTeam)
 
 	const bool justJoined = (GetTeamNumber() == TEAM_UNASSIGNED);
 
+	// Outside the below statement scope because this incorrectly triggers warnings on MSVC otherwise:
+	// https://developercommunity.visualstudio.com/t/fallthrough-does-not-suppress-warning-c26819-when/1206339
+	enum JoinMode {
+		Random = -2,
+		TeamWithLessPlayers = -1,
+	};
+
 	// Player bots should initially join a player team.
 	// Note that we can't do a ->IsBot check here, because the bot has not
 	// received its fakeclient flags yet at this point. Hence using the
 	// m_bNextClientIsFakeClient workaround.
 	if (justJoined && NEORules()->m_bNextClientIsFakeClient && !IsHLTV())
 	{
 		Assert(gpGlobals->curtime >= m_flNextTeamChangeTime);
-		iTeam = RandomInt(TEAM_JINRAI, TEAM_NSF);
+		
+		ConVarRef joinMode{ "bot_next_team" };
+
+		int numJin, numNsf;
+		switch (joinMode.GetInt())
+		{
+		case JoinMode::TeamWithLessPlayers:
+			numJin = GetGlobalTeam(TEAM_JINRAI) ? GetGlobalTeam(TEAM_JINRAI)->GetNumPlayers() : 0;
+			numNsf = GetGlobalTeam(TEAM_NSF) ? GetGlobalTeam(TEAM_NSF)->GetNumPlayers() : 0;
+			if (numJin < numNsf) {
+				iTeam = TEAM_JINRAI;
+				break;
+			}
+			else if (numNsf < numJin) {
+				iTeam = TEAM_NSF;
+				break;
+			}
+			[[fallthrough]];
+		case JoinMode::Random:
+			iTeam = RandomInt(TEAM_JINRAI, TEAM_NSF);
+			break;
+		default:
+			iTeam = joinMode.GetInt();
```

Comment:
Would it be worth defining an enum TOTAL_PLAYABLE_TEAMS or something and defaulting to the random behaviour if iTeam >= TOTAL_PLAYABLE_TEAMS ? Otherwise we hit a few asserts when passing a value greater than TEAM_NSF and the bots remain unassigned.

Reply:
Right, I didn't want to hard code a team limit here, however unlikely it is we'd ever have more. I've clamped the value between playable teams now.